src/lib/storage/partitioning/range_partition_schema.cpp

```diff
@@ -0,0 +1,88 @@
+#include <storage/partitioning/range_partition_schema.hpp>
+
+namespace opossum {
+
+RangePartitionSchema::RangePartitionSchema(ColumnID column_id, std::vector<AllTypeVariant> bounds)
+    : _column_id(column_id), _bounds(bounds) {
+  DebugAssert(std::all_of(bounds.cbegin(), bounds.cend(),
+                          [&bounds](const AllTypeVariant& each) { return each.which() == bounds.front().which(); }),
+              "All bounds have to be of the same type.");
+
+  _bound_type = data_type_from_all_type_variant(bounds.front());
+  _partitions.reserve(bounds.size() + 1);
+  for (PartitionID index{0}; index < bounds.size() + 1; ++index) {
+    _partitions.emplace_back(std::make_shared<Partition>(index));
+  }
+}
+
+std::string RangePartitionSchema::name() const { return "RangePartition"; }
+
+PartitionSchemaType RangePartitionSchema::get_type() const { return PartitionSchemaType::Range; }
+
+void RangePartitionSchema::append(const std::vector<AllTypeVariant>& values) {
+  AbstractPartitionSchema::append(values, get_matching_partition_for(values));
+}
+
+PartitionID RangePartitionSchema::get_matching_partition_for(const std::vector<AllTypeVariant>& values) const {
+  DebugAssert(values.size() > static_cast<size_t>(_column_id), "Can not determine partition, too few values given");
+  const auto& value = values[_column_id];
+  return get_matching_partition_for(value);
+}
+
+PartitionID RangePartitionSchema::get_matching_partition_for(const AllTypeVariant& value) const {
+  for (size_t index = 0; index < _bounds.size(); ++index) {
```

Comment:
I guess this method is heavily used -- did you consider to implement it with a logarithmic runtime (i.e. by doing a binary search?). Maybe keeping the bounds in a `std::set`, or using `std::lower_bound` on the vector could help.